state/storage.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+// ImmutableStorage implements [state.Immutable], allowing the map to be used
+// as a read-only state source, and making it compatible with error-generating backing storage.
```

Comment:
What does `"error-generating backing storage"` mean?

Perhaps:

```suggestion
// ImmutableStorage implements [state.Immutable] by wrapping a key-value map
```